migrations.go

```diff
@@ -33,20 +28,46 @@ func (m *Migration) String() string {
 	return strconv.FormatInt(m.Version, 10)
 }
 
+// A Group is a collection of migrations supporting registration and execution.
+type Group struct {
```

Comment:
Could you rename `Group` to either `Migrations` or `Collection`? It seems to be a much more natural name for this.

Reply:
I've opted for `Collection`, since having both `Migration` and `Migrations` types exposed by this package seemed confusing.